File: 백준/Silver/10828. 스택/스택.cc
#include<iostream>
#include<vector>

using namespace std;

class Stack {

	int* buffer;
	const int size;
	int top;
	int count;
public:
	Stack(int size) :size(size) {
		buffer = new int[size];
		top = -1;
		count = 0;
	}
	~Stack() {
		delete[] buffer;
	}

	bool Push(int data) {

		if (IsFull()) {
			return false;
		}
		++count;
		++top;
		buffer[top] = data;

		return true;
	}

	int Pop() {
		if (IsEmpty()) {
			return -1;
		}
		--count;
		int re = buffer[top];
		--top;

		return re;

	}

	int Top() {

		if (IsEmpty())
			return -1;
		else
			return buffer[top];
	}
	int Count() {
		return count;
	}


	int IsEmpty() {

		return top == -1;
	}

	int IsFull() {
		return (top + 1) == size;
	}



};

int main() {

	int N;
	string* str;
	int put_num;
	vector <int> num;
	int num_idx = 0;
	cin >> N;
	str = new string[N];
	
	Stack st(N);

	for (int i = 0; i < N; i++) {
		cin >> str[i];
		if (str[i] == "push") {
			cin >> put_num;
			num.push_back(put_num);
			
		}
		
	}
	/*
	cout << "--------여기서 부터 확인 입니다.------" << '\n';
	for (int i = 0; i < N; i++) {
		cout << str[i] << '\n';
	}


	cout << "--------여기서 부터 출력 입니다.------" << '\n';
	*/
	num_idx = 0;
	for (int i = 0; i < N; i++) {
		if (str[i] == "push") {
			st.Push(num[num_idx]);
			num_idx += 1;
		}
		else if (str[i] == "top") {
			cout << st.Top() << '\n';
		}
		else if (str[i] == "pop") {
			cout << st.Pop() << '\n';
		}
		else if (str[i] == "size") {
			cout << st.Count() << '\n';
		}
		else if (str[i] == "empty") {
			cout << st.IsEmpty() << '\n';
		}

	}


	return 0;
}

File: 백준/Silver/10828. 스택/스택.java

import java.io.*;
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.Stack;

public class Main {
    public static void main(String[] args) throws IOException {

        int N;
        Stack<String> stack = new Stack<>();

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        N = Integer.parseInt(br.readLine());

        for (int i=0; i<N; i++) {
            String[] str = br.readLine().split(" ");
            if (str[0].equals("push")) {
                stack.push(str[1]);
            }
            else {
                if (str[0].equals("pop")) {
                    if (!stack.isEmpty()) {
                        System.out.println(stack.pop());
                    }
                    else {
                        System.out.println(-1);
                    }
                }
                else if (str[0].equals("size")) {
                    System.out.println(stack.size());
                }
                else if (str[0].equals("empty")) {
                    if (!stack.isEmpty()) {
                        System.out.println(0);
                    }
                    else {
                        System.out.println(1);
                    }
                }
                else if (str[0].equals("top")) {
                    if (!stack.isEmpty()) {
                        System.out.println(stack.peek());
                    }
                    else {
                        System.out.println(-1);
                    }
                }
            }
        }


//        System.out.println("Hello world!  " + N);
    }
}